### PR TITLE
PFR-1219 : make chat responsive to smartphones

### PR DESCRIPTION
**Pull request by LamiaBouhajja**

### CODE REVIEW

File: application/src/main/webapp/js/chat.js [hunk 1]
```diff
@@ -178,6 +190,36 @@ var chatApplication = new ChatApplication();
...
+        $("#submit").on("click", function() {
+
+            var msg = $("#msg").val();
+
+              //    console.log("keyup : "+event.which + ";"+msg.length+";"+keydown);
+                  if ( msg.trim().length>=1) {
+                    chatApplication.sendMessage(msg);
+
+                  }
+                  // UP Arrow
```
Comment (thomasdelhomenie) [inline application/src/main/webapp/js/chat.js]: This is duplicated code. It should be factorized.

File: application/src/main/java/org/exoplatform/chat/portlet/chat/templates/index.gtmpl [hunk 2]
```diff
@@ -21,56 +21,159 @@
...
+                  <div class="buttonChangeStatus actionIcon dropdown" rel="tooltip" data-placement="bottom" data-toggle="tooltip" title="&{exoplatform.chat.change.status}"><span class="chat-status-chat chat-status-invisible" style="padding: 0;margin: 0;"></span></div>
+                  <ul class="dropdown-menu chat-status-panel">
+                      <li class="chat-menu" status="available"><a href="javaScript:void(0)"><i class="chat-status chat-status-available"></i>&{exoplatform.chat.available}</a></li>
+                      <li class="chat-menu" status="away"><a href="javaScript:void(0)"><i class="chat-status chat-status-away"></i>&{exoplatform.chat.away}</a></li>
+                      <li class="chat-menu" status="donotdisturb"><a href="javaScript:void(0)"><i class="chat-status chat-status-donotdisturb"></i>&{exoplatform.chat.donotdisturb}</a></li>
+                      <li class="chat-menu" status="invisible"><a href="javaScript:void(0)"><i class="chat-status chat-status-invisible"></i>&{exoplatform.chat.invisible}</a></li>
+                  </ul>
+              </div>
+
+              <label class="pull-left" style="margin-top: 27px;margin-bottom: 0; width: 60%;">
```
Comment (thomasdelhomenie) [inline application/src/main/java/org/exoplatform/chat/portlet/chat/templates/index.gtmpl]: Avoid inline css

File: application/src/main/java/org/exoplatform/chat/portlet/chat/templates/index.gtmpl [hunk 3]
```diff
@@ -21,56 +21,159 @@
...
+                  <% if ("DEV".equals(demoMode)) { %>
+                  <span class="label-demo">(Demo Mode)</span>
+                  <% } %>
+              </label>
+              <hr class="clearfix"/>
+              <span class="user_mail">email@exoplatform.com</span>
+          </div>
+          <div class="extraLeftMedium">
+              <div id="configButton" class="uiActionWithLabel ligne" data-placement="bottom" data-toggle="tooltip" title="&{exoplatform.chat.settings.button.tip}">
+                  <i class="uiIconSetting uiIconLightGray" style="margin-right: 10px;"></i> Preferences
```
Comment (thomasdelhomenie) [inline application/src/main/java/org/exoplatform/chat/portlet/chat/templates/index.gtmpl]: Must be internationalized

File: application/src/main/java/org/exoplatform/chat/portlet/chat/templates/index.gtmpl [hunk 4]
```diff
@@ -21,56 +21,159 @@
...
+
+              <label class="pull-left" style="margin-top: 27px;margin-bottom: 0; width: 60%;">
+                  <a href='/portal/intranet/profile/<%=user%>' class='label-user' target='_blank'><%=(fullname.length()>29)?fullname.substring(0,29)+"...":fullname%></a>
+                  <!--<span class="label-user-title" id="topTitle">Chat</span>-->
+                  <% if ("DEV".equals(demoMode)) { %>
+                  <span class="label-demo">(Demo Mode)</span>
+                  <% } %>
+              </label>
+              <hr class="clearfix"/>
+              <span class="user_mail">email@exoplatform.com</span>
```
Comment (thomasdelhomenie) [inline application/src/main/java/org/exoplatform/chat/portlet/chat/templates/index.gtmpl]: email hardcoded

File: application/src/main/java/org/exoplatform/chat/portlet/chat/assets/base.less [hunk 5]
```diff
@@ -1028,3 +1046,25 @@ footer
...
+#searchButton2{top : 13px}
+#menuButton.displayContent{
+    left: 0;
+    right: 0
+}
+#chat-status{
+    display: block!important;
+}
+.UIToolbarContainer .UserInfoPortletTDContainer [class^="uiIconPLF24x24"], .UIToolbarContainer .UIHelpPlatformToolbarPortletTDContainer [class^="uiIconPLF24x24"], .UIToolbarContainer .SearchPortletTDContainer [class^="uiIconPLF24x24"], .UIToolbarContainer .UICreatePlatformToolBarPortletIDTDContainer [class^="uiIconPLF24x24"], .UIToolbarContainer .UISetupPlatformToolBarPortletIDTDContainer [class^="uiIconPLF24x24"], .UIToolbarContainer .NotificationPopoverPortletTDContainer [class^="uiIconPLF24x24"], .UIToolbarContainer .AdminToolBarPortletTDContainer [class^="uiIconPLF24x24"], .UIToolbarContainer .middle-topNavigation-containerTDContainer [class^="uiIconPLF24x24"], .UIToolbarContainer .right-topNavigation-containerTDContainer [class^="uiIconPLF24x24"]{
+        @media (max-width: 599px) {
```
Comment (LamiaBouhajja) [inline application/src/main/java/org/exoplatform/chat/portlet/chat/assets/base.less]: I have to move this code to the platform-ui, it concerne the chat icon of the home page which is hidden on responsive view

File: application/src/main/java/org/exoplatform/chat/portlet/chat/assets/base.less [hunk 6]
```diff
@@ -1,5 +1,5 @@
 // Base CSS
-#configButton {
+#configButton, #configButton1 {
```
Comment (thomasdelhomenie) [inline application/src/main/java/org/exoplatform/chat/portlet/chat/assets/base.less]: configButton1 is not meaningful, please rename it.

File: application/src/main/java/org/exoplatform/chat/portlet/chat/assets/base.less [hunk 7]
```diff
@@ -214,9 +223,18 @@
...
+    -ms-transition-duration: .5s;
+    -moz-transition-duration: .5s;
+    -webkit-transition-duration: .5s;
+    transition-duration: .5s;
+    &.hideContent{opacity: 0; display: none}
+    &.displayContent{opacity: 1; display: block}
 }
 .menu-standard {
   display: inline;
+  i{font-size: 16px;
```
Comment (thomasdelhomenie) [inline application/src/main/java/org/exoplatform/chat/portlet/chat/assets/base.less]: Indentation is not good.

File: application/src/main/java/org/exoplatform/chat/portlet/chat/assets/responsive-400px-max.less [hunk 8]
```diff
@@ -17,98 +17,7 @@
...
-          }
-
-          .room-user-name, .room-user-status {
-            display: none;
-          }
-        }
-      }
-    }
-  }
+ searchButton
```
Comment (thomasdelhomenie) [inline application/src/main/java/org/exoplatform/chat/portlet/chat/assets/responsive-400px-max.less]: Isn't there a missing # ?

File: application/src/main/java/org/exoplatform/chat/portlet/chat/templates/index.gtmpl [hunk 9]
```diff
@@ -20,57 +20,162 @@
   <div class="chat-login-panel" style="display:none;"></div>
   <div class="chat-about-panel" style="display:none;"></div>
 <% if (!"public".equals(view)) { %>
+  <meta name="viewport" content="width=device-width, initial-scale=1 maximum-scale=1, user-scalable=no">
```
Comment (thomasdelhomenie) [inline application/src/main/java/org/exoplatform/chat/portlet/chat/templates/index.gtmpl]: I thought we said we should not prevent the zoom, no ?

File: application/src/main/java/org/exoplatform/chat/portlet/chat/templates/index.gtmpl [hunk 10]
```diff
@@ -20,57 +20,162 @@
...
+                <li class="chat-menu" status="away"><a href="javaScript:void(0)"><i class="chat-status chat-status-away"></i>&{exoplatform.chat.away}</a></li>
+                <li class="chat-menu" status="donotdisturb"><a href="javaScript:void(0)"><i class="chat-status chat-status-donotdisturb"></i>&{exoplatform.chat.donotdisturb}</a></li>
+                <li class="chat-menu" status="invisible"><a href="javaScript:void(0)"><i class="chat-status chat-status-invisible"></i>&{exoplatform.chat.invisible}</a></li>
+            </ul>
+    </div>
 
       <label class="pull-left" style="margin-top: 12px;">
-        <a href='/portal/intranet/profile/<%=user%>' class='label-user' target='_blank'><%=(fullname.length()>29)?fullname.substring(0,29)+"...":fullname%></a>
+        <a href='/portal/intranet/profile/<%=user%>' class='label-user' target='_blank'>&{exoplatform.chat.discussion}</a>
+        <!--<span class="label-user-title" id="topTitle">Chat</span>-->
```
Comment (thomasdelhomenie) [inline application/src/main/java/org/exoplatform/chat/portlet/chat/templates/index.gtmpl]: To be removed

File: application/src/main/java/org/exoplatform/chat/portlet/chat/templates/index.gtmpl [hunk 11]
```diff
@@ -20,57 +20,162 @@
...
+    </span>
+    <span class="menu-standard pull-center">
+      <!--<span class="avatar-user">
         <div class="activityAvatar avatarCircle" >
           <a rel="tooltip" data-placement="bottom" href="/portal/intranet/profile/<%=user%>" data-original-title="">
             <img class="avatar-image" onerror="this.src='/chat/img/user-default.jpg;'" src='/rest/chat/api/1.0/user/getAvatarURL/<%=user%>' >
           </a>
         </div>
-      </span>
+      </span>-->
```
Comment (thomasdelhomenie) [inline application/src/main/java/org/exoplatform/chat/portlet/chat/templates/index.gtmpl]: To be removed

File: application/src/main/java/org/exoplatform/chat/portlet/chat/templates/index.gtmpl [hunk 12]
```diff
@@ -20,57 +20,162 @@
...
+                      <li class="chat-menu" status="available"><a href="javaScript:void(0)"><i class="chat-status chat-status-available"></i>&{exoplatform.chat.available}</a></li>
+                      <li class="chat-menu" status="away"><a href="javaScript:void(0)"><i class="chat-status chat-status-away"></i>&{exoplatform.chat.away}</a></li>
+                      <li class="chat-menu" status="donotdisturb"><a href="javaScript:void(0)"><i class="chat-status chat-status-donotdisturb"></i>&{exoplatform.chat.donotdisturb}</a></li>
+                      <li class="chat-menu" status="invisible"><a href="javaScript:void(0)"><i class="chat-status chat-status-invisible"></i>&{exoplatform.chat.invisible}</a></li>
+                  </ul>
+              </div>
+
+              <label class="pull-left full_name">
+                  <a href='/portal/intranet/profile/<%=user%>' class='label-user' target='_blank'><%=(fullname.length()>29)?fullname.substring(0,29)+"...":fullname%></a>
+                  <!--<span class="label-user-title" id="topTitle">Chat</span>-->
```
Comment (thomasdelhomenie) [inline application/src/main/java/org/exoplatform/chat/portlet/chat/templates/index.gtmpl]: To be removed

File: application/src/main/java/org/exoplatform/chat/portlet/chat/templates/index.gtmpl [hunk 13]
```diff
@@ -20,57 +20,162 @@
...
+          <div class="extraLeftMedium">
+              <div id="configButton" class="uiActionWithLabel ligne" data-placement="bottom" data-toggle="tooltip" title="&{exoplatform.chat.settings.button.tip}">
+                  <i class="uiIconSetting uiIconLightGray" style="margin-right: 10px;"></i> &{exoplatform.chat.settings.button.tip}
+              </div>
+              <div id="tribeButton" class="uiActionWithLabel ligne" data-placement="home" data-toggle="tooltip" title="&{exoplatform.chat.settings.home}">
+                  <a href="/" ><i class="uiIconHomeInfo uiIconLightGray" style="margin-right: 10px;"></i> &{exoplatform.chat.home}</a>
+              </div>
+              <!-- <div id="tribeButton" class="uiActionWithLabel ligne" data-placement="bottom" data-toggle="tooltip" title="&{exoplatform.chat.settings.button.tip}">
+                  <i class="uiIconInformation uiIconLightGray" style="margin-right: 10px;"></i> About
+              </div>-->
```
Comment (thomasdelhomenie) [inline application/src/main/java/org/exoplatform/chat/portlet/chat/templates/index.gtmpl]: To be removed

File: application/src/main/webapp/js/chat.js [hunk 14]
```diff
@@ -368,20 +381,57 @@ var chatApplication = new ChatApplication();
...
+            }
+        }
+        chatApplication.loadRoom();
     });
 
     $(document).on("click", "#close-room-notif-config", function() {//close the setting page and go for the previous screen
-        chatApplication.loadRoom();
+
+         if(window.innerWidth > 767){
+                  chatApplication.loadRoom();
```
Comment (thomasdelhomenie) [inline application/src/main/webapp/js/chat.js]: indentation is not good

File: application/src/main/webapp/js/chat.js [hunk 15]
```diff
@@ -692,26 +806,20 @@ var handleRoomNotifLayout = function() {
...
 
-    $(".room-detail-fullname").on("click", function() {
-      if (chatApplication.isMobileView()) {
-        $(".uiLeftContainerArea").css("display", "block");
-        $(".uiRightContainerArea").css("display", "none");
-      }
     });
 
-    $('#chat-search').keyup(function(event) {
+    $('#chat-search, #chat-search2').keyup(function(event) {
```
Comment (thomasdelhomenie) [inline application/src/main/webapp/js/chat.js]: chat-search2 is not meaningful, please change it.

File: application/src/main/webapp/js/chat.js [hunk 16]
```diff
@@ -2432,8 +2604,11 @@ ChatApplication.prototype.loadRoom = function() {
     ////// USER
     {
       jqchat(".uiRoomUsersContainerArea").hide();
+      jqchat(".meeting-action-task").css("display", "none");
+//      jqchat(".meeting-actions").css("display", "none");
```
Comment (thomasdelhomenie) [inline application/src/main/webapp/js/chat.js]: To be removed

File: application/src/main/webapp/js/chat.js [hunk 17]
```diff
@@ -2983,16 +3165,73 @@ ChatApplication.prototype.jQueryForUsersTemplate = function() {
...
+
+    jqchat(".uiGlobalRoomsContainer").addClass("hideContent").removeClass("displayContent");
+
+    setTimeout(function(){
+         jqchat(".uiGlobalRoomsContainer").css("display", "none");
+    }, 500);
+
+//    if(window.innerWidth <= 767){
+//            jqchat("#topTitle").text("discussions");
+//        }
```
Comment (thomasdelhomenie) [inline application/src/main/webapp/js/chat.js]: To be removed

File: application/src/main/java/org/exoplatform/chat/portlet/chat/assets/base.less (hunk 7 above)
Comment (LamiaBouhajja) [inline application/src/main/java/org/exoplatform/chat/portlet/chat/assets/base.less]: Done

**Pull request closed without merging**